File: backend/internal/handlers/stats.go
package handlers

import (
	"net/http"

	"github.com/go-chi/chi/v5"
)

// GET /api/channels/{id}/stats
func (h *Handlers) ChannelStats(w http.ResponseWriter, r *http.Request) {
	stats, err := h.Store.ChannelStats(r.Context(), chi.URLParam(r, "id"))
	if handleStoreErr(w, err) {
		return
	}
	writeJSON(w, http.StatusOK, stats)
}

// GET /api/admin/stats
func (h *Handlers) PlatformStats(w http.ResponseWriter, r *http.Request) {
	stats, err := h.Store.PlatformStats(r.Context())
	if handleStoreErr(w, err) {
		return
	}
	writeJSON(w, http.StatusOK, stats)
}

File: src/api/stats.ts
import { api } from './client'
import type { ChannelStats, PlatformStats } from '@/shared/types'

export async function getChannelStats(channelId: string): Promise<ChannelStats | null> {
  try {
    return await api<ChannelStats>(`/api/channels/${channelId}/stats`)
  } catch {
    return null
  }
}

export function getPlatformStats(): Promise<PlatformStats> {
  return api<PlatformStats>('/api/admin/stats')
}
